### PR TITLE
SC2: dynamically create Beat <mission_name> Events, preventing copy-paste errors

### PR DESCRIPTION
**Pull request by Berserker66**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
Victory and Beat conditions diverged in some instances due to needing to be updated in two places, despite them being the same "check".

## How was this tested?
running Generate a few times and peeking at spoiler.

@TheCondor07

### CODE REVIEW

File: worlds/sc2wol/Locations.py [hunk 1]
```diff
@@ -314,9 +258,15 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
...
-        LocationData("Shatter the Sky", "Beat Shatter the Sky", None,
-                     lambda state: state._sc2wol_has_competent_comp(world, player)),
         LocationData("All-In", "All-In: Victory", None)
     ]
 
-    return tuple(location_table)
+    beat_events = []
+
+    for location_data in location_table:
+        if location_data.name.endswith((": Victory", ": Defeat")):
```
Comment (Ziktofel) [inline worlds/sc2wol/Locations.py:267]: I'd prefer a method at LocationData class

File: worlds/sc2wol/Locations.py [hunk 2]
```diff
@@ -314,9 +258,15 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
...
+            beat_events.append(
+                location_data._replace(name="Beat " + location_data.name.rsplit(": ", 1)[0], code=None)
```
Comment (Ziktofel) [inline worlds/sc2wol/Locations.py:269]: A function o generate the name?

Comment (Berserker66) [reply worlds/sc2wol/Locations.py:269]: if there's at least two places where this code would get used, sure.

File: worlds/sc2wol/Locations.py (hunk 1 above)
Comment (Berserker66) [reply worlds/sc2wol/Locations.py:267]: technically that's %100 == 0, except for All-In which breaks that combo

File: worlds/sc2wol/Locations.py (hunk 2 above)
Comment (Ziktofel) [reply worlds/sc2wol/Locations.py:269]: Sometimes I do that even if isn't when it provides a better readibility

**Comment by TheCondor07** [comment]
Sorry for the delay, looks good!

**Review by black-sliver — APPROVED** [review]
I have not tested. Trusting Condor on that one :-)

**Pull request merged**